internal/aws/awsutil/conn.go

```diff
@@ -320,3 +371,15 @@ func getPartition(region string) string {
 	p, _ := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
 	return p.ID()
 }
+
+func CreateIMDSV2AndFallbackClient(cfg awsSDKV2.Config) (*imds.Client, *imds.Client) {
+	optionsIMDSV2Only := func(o *imds.Options) {
+		o.EnableFallback = awsSDKV2.FalseTernary
+	}
+	optionsIMDSV1Fallback := func(o *imds.Options) {
+		o.EnableFallback = awsSDKV2.TrueTernary
```

Comment:
We explicitly try IMDSV2, then we explicitly try IMDSV1. What would the fallback be after IMDSV1?

Reply:
No, we explicitly try imdsv2 then we allow a fallback. I do not think the sdk allows imdsv2 then retry v2 then fallback it has to be try v2 then have a client that will try both

Reply:
There is no option in the sdk to only reach out to imdsv1